manifests/server.pp

```diff
@@ -363,100 +363,101 @@
+  $post_hook_content               = $::puppet::server_post_hook_content,
+  $post_hook_name                  = $::puppet::server_post_hook_name,
+  $storeconfigs_backend            = $::puppet::server_storeconfigs_backend,
+  $app_root                        = $::puppet::server_app_root,
+  $ruby_load_paths                 = $::puppet::server_ruby_load_paths,
+  $ssl_dir                         = $::puppet::server_ssl_dir,
+  $ssl_dir_manage                  = $::puppet::server_ssl_dir_manage,
+  $ssl_protocols                   = $::puppet::server_ssl_protocols,
+  $package                         = $::puppet::server_package,
+  $version                         = $::puppet::server_version,
+  $certname                        = $::puppet::server_certname,
+  $enc_api                         = $::puppet::server_enc_api,
+  $report_api                      = $::puppet::server_report_api,
+  $request_timeout                 = $::puppet::server_request_timeout,
+  $ca_proxy                        = $::puppet::server_ca_proxy,
+  $strict_variables                = $::puppet::server_strict_variables,
+  $additional_settings             = $::puppet::server_additional_settings,
+  $rack_arguments                  = $::puppet::server_rack_arguments,
+  $foreman                         = $::puppet::server_foreman,
+  $foreman_url                     = $::puppet::server_foreman_url,
+  $foreman_ssl_ca                  = $::puppet::server_foreman_ssl_ca,
+  $foreman_ssl_cert                = $::puppet::server_foreman_ssl_cert,
+  $foreman_ssl_key                 = $::puppet::server_foreman_ssl_key,
+  $server_foreman_facts            = $::puppet::server_foreman_facts,
+  $puppet_basedir                  = $::puppet::server_puppet_basedir,
+  $puppetdb_host                   = $::puppet::server_puppetdb_host,
+  $puppetdb_port                   = $::puppet::server_puppetdb_port,
+  $puppetdb_swf                    = $::puppet::server_puppetdb_swf,
+  $parser                          = $::puppet::server_parser,
+  $environment_timeout             = $::puppet::server_environment_timeout,
+  $jvm_java_bin                    = $::puppet::server_jvm_java_bin,
+  $jvm_config                      = $::puppet::server_jvm_config,
+  $jvm_min_heap_size               = $::puppet::server_jvm_min_heap_size,
+  $jvm_max_heap_size               = $::puppet::server_jvm_max_heap_size,
+  $jvm_extra_args                  = $::puppet::server_jvm_extra_args,
+  $jruby_gem_home                  = $::puppet::server_jruby_gem_home,
+  $max_active_instances            = $::puppet::server_max_active_instances,
+  $max_requests_per_instance       = $::puppet::server_max_requests_per_instance,
+  $use_legacy_auth_conf            = $::puppet::server_use_legacy_auth_conf,
+  $check_for_updates               = $::puppet::server_check_for_updates,
+  $environment_class_cache_enabled = $::puppet::server_environment_class_cache_enabled,
 ) {
 
   validate_bool($ca)
```

Comment:
Also add a `validate_bool` for `$environment_class_cache_enabled` to the block around line 520 please.